### PR TITLE
dev-python/shiboken,dev-python/pyside,dev-python/pyside-tools: 5.11.2

### PR DESCRIPTION
**Pull request by Perlovka**

### CODE REVIEW

File: dev-python/shiboken/files/shiboken-5.11-fix-warnings.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- a/libshiboken/pep384impl.cpp
++++ b/libshiboken/pep384impl.cpp
+@@ -130,7 +130,7 @@ check_PyTypeObject_valid(void)
+     long probe_tp_dictoffset = PyLong_AsLong(d);
+     PyObject *probe_tp_mro = PyObject_GetAttrString(obtype, "__mro__");
+     if (false
+-        || probe_tp_name            != check->tp_name
++        || (probe_tp_name - check->tp_name) != 0 // to avoid warning
```
Comment (Pesa) [inline dev-python/shiboken/files/shiboken-5.11-fix-warnings.patch]: what was the warning? does it cause a build failure? if not, I'd rather avoid adding this patch.

**Comment by alexxy** [comment]
Any news about this change? I need this packages to make FreeCAD available in main tree

**Comment by a17r** [comment]
Qt for Python 5.12 was released: http://blog.qt.io/blog/2018/12/18/qt-python-5-12-released/

File: dev-python/shiboken/files/shiboken-5.11-fix-warnings.patch (hunk 1 above)
Comment (Perlovka) [reply dev-python/shiboken/files/shiboken-5.11-fix-warnings.patch]: * QA Notice: Package triggers severe warnings which indicate that it
 *            may exhibit random runtime failures.
 * /var/tmp/portage/dev-python/shiboken-5.11.2/work/pyside-setup-everywhere-src-5.11.2/sources/shiboken2/libshiboken/pep384impl.cpp:133:47: warning: comparison with string literal results in unspecified behavior [-Waddress]

The patch has been taken from qt bugs or forums, can't find this now
5.12 has slightly another solution http://code.qt.io/cgit/pyside/pyside-setup.git/tree/sources/shiboken2/libshiboken/pep384impl.cpp?h=v5.12.0&id=0997b2055007da282cd52087632a28645e3007d1#n133

Comment (Perlovka) [reply dev-python/shiboken/files/shiboken-5.11-fix-warnings.patch]: Here is origin: http://code.qt.io/cgit/pyside/pyside-setup.git/commit/?h=5.11.3&id=70d241713ae375124eaabdd928fcdfd034822c6e

**Comment by leycec** [comment]
@Pesa: There any significant blockers on @Perlovka's end for this? If not, perhaps we could gradually:

1. Merge this (whenever convenient for your scarce free time, of course).
1. Address the [PySide2 5.12 release](http://blog.qt.io/blog/2018/12/18/qt-python-5-12-released) with a follow-up PR to this overlay, authored either by @Perlovka or myself. If @Perlovka's occupied elsewhere, I might be able to lend my volunteer services here again.
1. Encourage Qt overlay users to test the stable release for a week or two. Naturally, I'll do so as well. I have a [several million-line PySide2-based multiphysics biology simulator](https://gitlab.com/betse/betsee) that certainly puts PySide2 through its paces.
1. Submit a new [gentoo/gentoo](https://github.com/gentoo/gentoo) PR merging these ebuilds into the Portage tree, authored by... *somebody.* This may not be me, due to work-related time constraints.

<sup>This has been a partial repost from [bug 624682](https://bugs.gentoo.org/624682#c58).</sup>

**Comment by leycec** [comment]
@Pesa: *please don't be dead*

**Comment by efferre79** [comment]
the current version in the overlay doesn't build against qt 5.12.3 due to the following error:

-- Configuring done
CMake Error at cmake/Macros/PySideModules.cmake:113 (add_dependencies):
  The dependency target "QtMultimedia" of target "QtTextToSpeech" does not
  exist.
Call Stack (most recent call first):
  PySide2/QtTextToSpeech/CMakeLists.txt:29 (create_pyside_module)

**Comment by leycec** [comment]
Sadly, @Pesa and hence this pull request appears to be effectively dead. There hasn't been *any* meaningful movement here for over six months.

To remedy this, we could consider hosting the PySide2 ebuilds elsewhere until ready for "primetime" Portage integration. As the original author of these ebuilds, [I run a third-party overlay](https://github.com/leycec/raiagent) that would be *delighted* to temporarily maintain these ebuilds. Likewise, I respond promptly to outstanding issues and pull requests. So, that might work.

Of course, the ideal scenario would be for Qt repository maintainers (e.g., @Pesa, @kensington, @a17r) to resurrect this pull request from the dead. I don't necessarily see that happening, but it certainly would be preferable to a friendly community fork. :fork_and_knife:

**Comment by Pesa** [comment]
Nothing is magically going to happen here unless someone steps up to become a proxy maintainer for this package (or set of packages). This has proven to be a non-trivial package to maintain and I have no motivation to work on it (nor spare time for that matter), and it is clear that no other current developer wants to take over maintenance.

> To remedy this, we could consider hosting the PySide2 ebuilds elsewhere until ready for "primetime" Portage integration. As the original author of these ebuilds, I run a third-party overlay that would be delighted to temporarily provide working versions of PySide2 2.0.0. Likewise, I respond promptly to outstanding issues and pull requests. So, that might work.

That's great! (btw, anyone can maintain an unofficial overlay, you don't need to ask us for permission, just go ahead and create one)

If you're willing to become proxy maintainer for pyside and related packages, that'd be even better, so let us know if you're interested.

**Comment by Pesa** [comment]
And no, it doesn't make sense to merge this PR. It's obsolete. The proper course of action is to update the `-9999` ebuilds so that they work with Qt 5.12 and later. Then fork a 5.12.x (or 5.13.x if it has already happened) from the `-9999` and let them bake in the overlay for some time. Eventually (and hopefully) we'll move the stable releases to the main gentoo repo.

**Comment by leycec** [comment]
> If you're willing to become proxy maintainer for pyside and related packages...

**Absolutely.** I actively maintain a [PySide2-based multiphysics biology simulator](https://gitlab.com/betse/betse) with [corresponding Gentoo ebuilds hosted at my overlay](https://github.com/leycec/raiagent/tree/master/sci-biology). I'm invested in this. Ensuring that PySide2 remains usable under Gentoo is a primary personal concern.

**tl;dr:** *I'd be happy to proxy maintain PySide2 until Portage support lands.*

> And no, it doesn't make sense to merge this PR. It's obsolete.

That's great to know. I thought there might be *something* of value to salvage from @Perlovka's efforts, but... I suppose that was just wishful thinking. `</sigh>`

> The proper course of action is to update the `-9999` ebuilds so that they work with Qt 5.12 and later.

*Right.*

> Then fork a 5.12.x (or 5.13.x if it has already happened) from the -9999 and let them bake in the overlay for some time.

*Right.*

> Eventually (and hopefully) we'll move the stable releases to the main gentoo repo.

*Right.* This is a sound plan.

Once I've copied all `dev-python/*` ebuilds from this overlay into my own and confirmed their compatibility with Qt ≤ 5.11.x, I'll drop another comment here. To avoid confusion, we might consider removing the entire `dev-python` directory from this overlay at that time.

Thanks for the prompt reply, @Pesa. Your commentary is a tremendous help.

**Comment by Pesa** [comment]
Awesome! You're hired! ;)

**Comment by Pesa** [comment]
> Once I've copied all dev-python/* ebuilds from this overlay into my own and confirmed their compatibility with Qt ≤ 5.11.x, I'll drop another comment here. To avoid confusion, we might consider removing the entire dev-python directory from this overlay at that time.

wait... I meant the other way around. You send PRs to *this* overlay (and later to the main gentoo repo) and I or any other @gentoo/qt developer can merge them. Also let me clarify another couple of things:

1. no need to keep compat with anything older than Qt 5.12 at this time (unless you really want to), just declare the deps correctly in the ebuild.
2. regarding the live `-9999` ebuilds, we don't *have* to have them, it's really up to you as maintainer. I'm perfectly fine with having just ebuilds for the official releases. So if you're not interested in maintaining the `-9999` ebuilds, feel free to drop them.

**Comment by leycec** [comment]
> You send PRs to this overlay (and later to the main gentoo repo) and I or any other @gentoo/qt developer can merge them.

Oh, sure! That's even better. Preserving PySide2 in the Qt repository is for the betterment of mankind.

> no need to keep compat with anything older than Qt 5.12 at this time

**Excellent.** Qt 5.12.x compatibility is my highest priority. For simplicity:

* My next pull request will probably update the `dev-python/*` package tree to require only Qt 5.12.x.
* My next-next pull request will probably generalize that to Qt ≥ 5.12.0 and thus support Qt 5.13.x, too.

> (unless you really want to)

:no_entry_sign: 

> So if you're not interested in maintaining the -9999 ebuilds, feel free to drop them.

**Excellent.** If this was my own overlay, I'd preserve the `-9999` ebuilds out of rotten stubbornness. In this case, live ebuilds are one more needless thing that will inevitably go wrong for end users. So, my next pull request will probably also transition us from the live PySide2 repositories to the quasi-stable PySide2 2.0.0~alpha1 release.

Let's do this, Team Qt™!

**Comment by leycec** [comment]
Pull request incoming tomorrow, as I've just finished `shiboken-5.12.3.ebuild`.

This must be the dawning of the Age of Aquarius. :sun_with_face:

**Comment by leycec** [comment]
By tomorrow, of course, I meant today.
